xbmc/addons/interfaces/Filesystem.cpp

```diff
@@ -300,11 +300,10 @@ char* Interface_Filesystem::get_cache_thumb_name(void* kodiBase, const char* fil
               __FUNCTION__, kodiBase, static_cast<const void*>(filename));
     return nullptr;
   }
-
-  Crc32 crc;
-  crc.ComputeFromLowerCase(filename);
-  std::string string = StringUtils::Format("%08x.tbn", static_cast<unsigned int>(crc));
-  char* buffer = strdup(string.c_str());
+  
+  auto crc = Crc32::ComputeFromLowerCase(filename);
+  std::string hex = StringUtils::Format("%08x.tbn", crc);
```

Comment:
Or `const auto hex` ;-)

Reply:
I generally avoid using auto for `std::string` as it's such a fundamental type. But I guess that's preference ;)

Reply:
I'm with phunkyfish, I think that by not hiding types understandability increases, though I trust the author weighs these factors and makes a good engineering decision

Reply:
In this particular case I would actually go with
```cpp
const uint32_t crc = Crc32::ComputeFromLowerCase(filename);
const auto hex = StringUtils::Format("%08x.tbn", crc);
```
because the return type of `Crc32::ComputeFromLowerCase(filename);` is less obvious than the one of `StringUtils::Format("%08x.tbn", crc)`. If `StringUtils::Format()` doesn't return an `std::string` (or another string-like object) someone made some very bad choices when writing that method.

But this is getting off topic. My initial point was: If we use (`const`) `auto crc` one line above we should also use (`const`) `auto hex`.

Reply:
Agreed, both should be const.